crates/communication/src/server/parameters/subscriptions.rs

```diff
@@ -131,7 +131,33 @@ async fn handle_request<Parameters>(
                 }
             };
 
-            respond(request, response).await;
+            let data = {
+                let parameters = parameters_reader.next();
+                parameters.serialize_path::<TextualSerializer>(path)
+            };
+            let data = match data {
+                Ok(data) => data,
+                Err(error) => {
+                    respond(
+                        request,
+                        ParametersResponse::Subscribe {
+                            id,
+                            result: Err(format!("failed to serialize: {error:?}")),
+                        },
+                    )
+                    .await;
+                    return;
+                }
+            };
+            respond(request.clone(), response).await;
+            respond(
+                request,
+                ParametersResponse::SubscribedData {
+                    subscription_id: id,
+                    data,
+                },
+            )
+            .await
```

Comment:
Did you think about the order? One could argue that sending subscribed data belongs to the subscription request and we should therefore send the acknowledgment after sending the subscribed data. Could this lead to funny races in the way you implemented it currently?

Reply:
I think it depends on the exact logic the client expects, I can see arguments preferring either way.
- For update then confirmation, one might argue that after getting subscription confirmation data should be available immediately, i.e. everything should be set up before receiving the success message.
- For confirmation then update, one might argue that the confirmation belongs more closely to the request than the update does, since the confirmation is only sent once, whereas the update could be sent over and over again. This is also the same way output subscriptions work. The only difference there is, that we are not explicitly sending an initial update after subscribing because there'll be one after a few milliseconds anyways.